torch/nn/modules/activation.py

```diff
@@ -32,7 +34,7 @@ class Threshold(Module):
         >>> print(m(input))
     """
 
-    def __init__(self, threshold, value, inplace=False):
+    def __init__(self, threshold=0.0, value=0.0, inplace=False):
```

Comment:
Hmm can we keep the interface as it was? The default one is just ReLU, and it's clearer to use that module instead of `nn.Threshold()`

Reply:
This one, I'll push back on a little bit harder. It's important to be consistent on function signatures, especially in a language with duck typing, and especially in a field where one might want to introspect code to e.g. optimize hyperparameters or visualize what's going on.

Before this change, the nonlinear activation functions had different function signatures. In particular, it was impossible for a program to call `Threshold` without human intervention since there was no reasonable set of default values provided.

Use cases:

* The script which generated the images is oblivious to anything about the functions other that they are a nonlinear functions with one input and one output
* Optimizing hyperparameters, for example picking choice of nonlinearity through something like simulated annealing or genetic algorithms (and further optimizing parameter choices on those nonlinearities from a reasonable starting-point if desired).
* For learning purposes, having tools where students might have a visualization of these functions, and tweak parameters.

The next step would be to register these in some way (e.g. Python entry points).

Reply:
That said, I agree a human user should use `ReLU` when that's what's meant. I guess I have faith in people to do that without enforcing it :)

Reply:
These aren't really reasonable default values anyways. And I don't think I get the points 2 and 3. For 1, I would much rather change the script (e.g. maybe make a dict from function name to args, and print args in plot title), than change the signature of this.

Reply:
I agree with @SsnL. Threshold is really a family of functions, and you need to choose the one you want explicitly. If you want the defaults (which are indeed commonly used), we picked it already for you under the name of `nn.ReLU`. In the hyperparameter optimization case, if you want to limit yourself to parameter-less nonlinearities, you should just exclude Threshold in favor of ReLU.

Reply:
Thank you.

I removed Threshold from the PR. Given that the goals of documentation, I don't think an image with unreasonable defaults makes sense to include. My preference would be to add reasonable defaults for Threshold, but I think that's a separate discussion and PR, and a more limited change might makes more sense in this one.

I will respond to feedback to specific comments independently, since it may be moot.

Reply:
I'm responding to points above, but all of this is likely academic at this point, so please feel free to skip.

re: Points 2, 3.

In retrospect, that's clearly outside of the scope of this PR. It was a mistake on my part to raise them here. I'm glad to go into more depth if desired.

re: `Threshold`.

I think a key issue is that the terminology might be a little ambiguous. In classical neural networks literature, 'Threshold' generally referred to a step function:

![image](https://user-images.githubusercontent.com/1427775/36841530-9054dcb2-1d16-11e8-8357-ad3eb8230d87.png)

([source](http://staff.itee.uq.edu.au/janetw/cmc/chapters/Introduction/): a random text from 1997)

Other [references](http://slideplayer.com/slide/4799311/) refer to *any* non-linear activation as a threshold function. At the time, thresholds generally were, as you stated, non-zero. A Google Images search 'threshold function deep learning' and similar brings up zero plots as used in pytorch, so this is likely to be a point-of-confusion.

Ideally, the documentation might explain:

* What the function does at-a-glance;
* Points of confusion around terminology;
* Why it does what it does; and
* When it ought to be used

I think that's especially important with a function like this one, where:

* in networks with bias terms, `Threshold` might often behave just like a `ReLU` (perhaps with slightly different initial conditions)
* there is a historical body of texts using the terminology differently

The target of the documentation are people who might not already be familiar with this.

re: Plots with args

My goal with the plots was to give information-at-a-glance. With that, I prefer less clutter. If I were putting more time into this, I'd probably go in the other direction:

* Shrink the plots a little bit;
* Add x=0, y=0 axes (faint);
* Reduce the number of labelled ticks; and
* Possibly, remove title entirely, since it is clear from the context

Alternatively, I might clearly label the parameters on the image (e.g. an arrow pointing to the threshold, labelling it as such).

If more clarity on how parameters influence things is desired, interactives where one can play with the parameters might be the next step.